pkg/kube/krt/metrics.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package krt
+
+import (
+	"sync"
+
+	"istio.io/istio/pkg/env"
+	"istio.io/istio/pkg/monitoring"
+)
+
+var (
+	EnableKRTMetrics = env.Register("ISTIO_ENABLE_KRT_METRICS", true,
+		"If enabled, publishes metrics for queue depth, latency and processing times.").Get()
+
+	collectionTag = monitoring.CreateLabel("collection")
+	dependencyTag = monitoring.CreateLabel("dependency")
+	enableMetric  = monitoring.WithEnabled(func() bool {
+		return EnableKRTMetrics
+	})
+
+	events           = monitoring.NewSum("krt_incoming_event_count", "Count of events", enableMetric)
+	eventsSize       = monitoring.NewSum("krt_incoming_event_size", "Size of events", enableMetric)
+	eventsDownstream = monitoring.NewSum("krt_outgoing_event_size", "Size of outgoing events", enableMetric)
```

Comment:
I donot think the above three metrics make much sense. Mostly we donot care how many events processed in underlying lib. User may need it for particular bussiness handling